docs/data/data-grid/column-groups/column-groups.md

```diff
@@ -1,18 +1,90 @@
 ---
-title: Data Grid - Column groups 🚧
+title: Data grid - Column groups
 ---
 
-# Data grid - Column groups 🚧
+# Data grid - Column groups
 
 <p class="description">Group your columns.</p>
 
+Grouping columns allows you to have multiple levels of columns in your header and to toggle column groups to show and hide additional columns.
+
+## Define column grouping
+
+You can define the column grouping structure by providing the `columnGroupingModel` prop to the grid.
+This prop receives an array of column groups.
+
+A column group is defined by at least two properties:
+
+- `groupId`: a string used to identify the group
+- `children`: an array containing the children of the group
+
+The `children` attribute can contain two types of objects:
+
+- leafs with type `{ field: string }`, which add the column with the corresponding `field` to this group.
```

Comment:
Would it be more consistent with the rest of the terminology to use `{ column: string }` instead of `{ field : string }`?
How big would the impact of such a change be right now?

```suggestion
- leaves with type `{ column: string }`, which add columns with the corresponding names to this group.
```

Reply:
It's consistent with `field` in column definition: https://mui.com/x/react-data-grid/column-definition/

Reply:
But `field` refers to a field in data (row), while column groups refer to a GridColdef?

Reply:
In the column definition, `field` is required and must be unique. Then as a developer, I consider it as a nickname of `column_id`

Reply:
```suggestion
- leaves with type `{ field: string }`, which add the column with the corresponding `field` to this group.
```

Reply:
I think we should consider renaming `field` in `columnGroupingModel`.
While it feels consistent with `GridColDef['field']`, it doesn't refer to rows model (as opposed to `GridColDef['field']`).

Maybe use `columnField` instead?
We do this already in `GridFilterModel`: https://github.com/mui/mui-x/blob/291bd2684f619fbbef86ac3b47a069ba91c1cc48/packages/grid/x-data-grid/src/models/gridFilterItem.ts#L10-L13

Reply:
The `children` property contains the children of the group which are either a group, or a column. So having `{ field: "col1" }` means in my children I have a column with field `"col1"`. With `{ columnField: "col1" }` we understand what is `columnField` but what is the meaning of the object wrapping it?

In such a case, maybe it's better to replace `children` property by two distinct ones:
- `childrenColumnFields`: an array of column fields
- `childrenGroups`: an array of groups

Reply:
> With { columnField: "col1" } we understand what is columnField but what is the meaning of the object wrapping it?

Maybe I don't get your point here, but for me it's the same as with `{ field: "col1" }` - wrapper object is just an object, and its content would define whether it's a column or column group.
We might as well just use a string `"col1"` instead of `{ columnField: "col1" }`, but having object with a field is more descriptive.
So I don't think that renaming `field` to `columnField` should require any deeper changes in the model.

Reply:
By the way, the object is strange but latter on with the expansion it should have other options like `{ columnField: 'col1', mode: "expended"}` to say that with expended mode the col1 must be visible

Reply:
Overall, I think that `columnField` is a bit better because then at least it's clear that's not exclusively related to a data field.
Also, if we eventually decide to group within the column definitions, it'd probably be less confusing regarding what the prop is referencing.

On other hand, it's not a big difference, and the `column` prefix might be redundant most of the time. 
So if @alexfauquette feels `field` is better I'm fine with it.
We do need to fix the plural on `leaf` though.

Reply:
+1 for `columnField` for consistency with
https://github.com/mui/mui-x/blob/291bd2684f619fbbef86ac3b47a069ba91c1cc48/packages/grid/x-data-grid/src/models/gridFilterItem.ts#L10-L13

Reply:
I was in favor of renaming `columnField` => `field` in `GridFilterItem` :laughing: 

But we should definitely have the same on in both interfaces.
And in other as well. My point for renaming was that we have `field` in tons of interfaces except in `GridFilterItem` (`GridSortCellParams`, `GridColumnOrderChangeParams`, `GridEditCellValueParams`, `GridColumnHeaderParams`, ... to give a few)

Reply:
It's not exactly the same, because all the other interfaces are parameters, whereas `GridFilterItem` is a definition. But in each parameter, we use both `field` and `colDef` (which contains the field). Moreover we have the `sortModel=[{field: ...}]`

That explains why I'm used to referring to a given column by using `field`. It's what's done in all the `onXxxChange`. So I prefer to keep my first intuition, now that I'm sure it does not comes from nowhere